File: technical_skills_java/src/main/java/com/natiqhaciyef/technical_skills_java/leetcode/interview/BestTimeToBuyAndSellStock.java
package com.natiqhaciyef.technical_skills_java.leetcode.interview;

public class BestTimeToBuyAndSellStock {

    public int maxProfit(int[] prices) {
        int diff = 0;
        int min = prices[0];

        for (int i = 0; i < prices.length; i++) {
            if (i != prices.length - 1 && prices[i] < min) {
                min = prices[i];
            }

            if (prices[i] - min > diff) {
                diff = prices[i] - min;
            }
        }


        return diff;
    }

}

File: technical_skills_kotlin/src/main/java/com/natiqhaciyef/technical_skills_kotlin/leetcode/easy/Best Time to Buy and Sell Stock.kt
package com.natiqhaciyef.technical_skills_kotlin.leetcode.easy

fun maxProfit(prices: IntArray): Int {
    var diff = 0
    var min = prices[0]

    //    Constraints:
    //    1 <= prices.length <= 105
    //    0 <= prices[i] <= 104
    //    O(N) -> runtime of function

    for (index in prices.indices) {
        if (prices[index] < min && index != prices.lastIndex){
            min = prices[index]
            //            println("Min: $min")
        }

        if (prices[index] - min > diff){
            diff = prices[index] - min
            //            println("Diff: $diff")
        }
    }

    return diff
}